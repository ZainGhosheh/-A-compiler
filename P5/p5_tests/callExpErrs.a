f : (a: int, b: bool) -> int {

}

e : () -> void {

}

x : custom {

};

z : () -> void {
    #x(); # non-function 
    a:int = 1;
    b:bool = false;
    e();
    f(a, b);
    f(a);
    f(b, a);
    x();
}
