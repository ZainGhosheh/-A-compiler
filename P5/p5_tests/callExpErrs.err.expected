FATAL [19,5]-[19,9]: Function call with wrong number of args
FATAL [20,5]-[20,12]: Type of actual does not match type of formal
FATAL [20,5]-[20,12]: Type of actual does not match type of formal
FATAL [21,5]-[21,8]: Attempt to call a non-function
Type Analysis Failed
